git-branchless/src/revset/eval.rs

```diff
@@ -204,14 +207,35 @@ pub(super) fn eval_name(ctx: &mut Context, name: &str) -> EvalResult {
 }
 
 pub(super) fn eval_fn(ctx: &mut Context, name: &str, args: &[Expr]) -> EvalResult {
-    let function = FUNCTIONS
-        .get(name)
-        .ok_or_else(|| EvalError::UnboundFunction {
-            name: name.to_owned(),
-            available_names: FUNCTIONS.keys().sorted().copied().collect(),
-        })?;
-
-    function(ctx, name, args)
+    if let Some(function) = FUNCTIONS.get(name) {
+        return function(ctx, name, args);
+    }
+
+    let alias_key = format!("branchless.revsets.alias.{}", name);
+    let alias_template = ctx
+        .repo
+        .get_readonly_config()
+        .map_err(EvalError::OtherError)?
+        .get::<String, String>(alias_key)
+        .map_err(EvalError::OtherError)?;
+    if let Some(alias_template) = alias_template {
+        let alias_expr = parse(&alias_template)
+            .wrap_err("Parsing alias expression")
+            .map_err(EvalError::OtherError)?;
```

Comment:
Can you add a test to track the behavior for when an alias has a parse expression? In particular, we should probably surface the alias value in the error message.

Reply:
Thank you for this suggestion; the output is much more helpful now. Here's an example:
```
❯ git config branchless.revsets.alias.foo 'bar('
❯ cargo run -- query 'foo()'
Evaluation error for expression 'foo()': Failed to parse alias expression 'bar('
parse error: Unrecognized EOF found at 4
Expected one of "(", ")", "..", ":", "::", a commit/branch/tag or a string literal
```

Note that I used `eyre` to manually wrap and format the `ParseError` ... sorry to be dense about this, but I'm still feeling shaky about how to properly handle errors in some cases. Does this sound/look OK to you?

Reply:
Ideally we'll create new error types for reach failure case. You can see the documentation for [`thiserror`] for how to do it. In this case, we could create `EvalError::ParseRevsetError(ParseError)` to wrap the underlying type directly, rather than create a new `eyre::Error`.

Reply:
Ah, of course. Yes, that makes sense. Done! Thank you!